overlord/ifacestate/handlers.go

```diff
@@ -1039,7 +1044,32 @@ func batchConnectTasks(st *state.State, snapsup *snapstate.SnapSetup, conns map[
 	if len(ts.Tasks()) > 0 {
 		ts.AddTask(setupProfiles)
 	}
-	return ts, nil
+	return ts, hasHooks, nil
+}
+
+// firstTaskAfterBootWhenPreseeding finds the first task to be run for thisSnap
+// on first boot after mark-preseeded task, this is normally install hook.
+func firstTaskAfterBootWhenPreseeding(thisSnap string, markPreseeded *state.Task) (*state.Task, error) {
```

Comment:
maybe I'm misreading the doc-comment here, but my reading of this function is that it _always_ returns the install hook task for the given snap, so it's not really a generic "give me whatever the first task after boot is for this snap when preseeding" function it's more like "give me the install hook task for this snap using the mark preseeded task". I think the name of the function or at least the doc-comment should be updated to make clear that this only will ever return a task that corresponds to running the install hook, because I just assumed before reading the function implementation that this might reasonably `return nil, nil` if there was no install hook task for `thisSnap`, but in that case it will `return nil, "internal error: cannot find start-setup-profiles or install hook for snap %q"`

Reply:
You're right, this function evolved slightly over time, we've really expecting install hook task here and it's an internal error if it's missing. Install hook task is always created for newly installed snap, which is the case when preseeding.

Reply:
I think the name is fine now that we have the doc-comment on it that says it always returns the install hook task right now